src/components/pages/doc.js

```diff
@@ -199,27 +202,32 @@ function Doc({ initialState, PageTemplate }) {
     </div>
   );
 
-  const alerts = useGlobalAlerts();
-  if (state.currentDocOrRevision.archived) {
-    alerts.push({
-      message: t('common:archivedAlert'),
-      type: ALERT_TYPE.warning,
-      showInFullScreen: false
-    });
-  }
+  useEffect(() => {
+    const newAlerts = [...globalAlerts];
 
-  if (isExternalDocument) {
-    alerts.push({
-      message:
-        (<Trans
-          t={t}
-          i18nKey="common:externalDocumentWarning"
-          components={[<a key="external-document-warning" href={state.currentDocOrRevision.originUrl} />]}
-          />),
-      type: 'warning',
-      showInFullScreen: false
-    });
-  }
+    if (state.currentDocOrRevision.archived) {
+      newAlerts.push({
+        message: t('common:archivedAlert'),
+        type: ALERT_TYPE.warning,
+        showInFullScreen: false
+      });
+    }
+
+    if (isExternalDocument) {
+      newAlerts.push({
+        message:
+          (<Trans
+            t={t}
+            i18nKey="common:externalDocumentWarning"
+            components={[<a key="external-document-warning" href={state.currentDocOrRevision.originUrl} />]}
+            />),
+        type: 'warning',
+        showInFullScreen: false
+      });
+    }
+
+    setAlerts(newAlerts);
+  }, [globalAlerts, state.currentDocOrRevision, isExternalDocument, t]);
```

Comment:
Good that you also catched that one!

Reply:
I checked all `globalAlert` cases 😬 They all would have had the same bug.